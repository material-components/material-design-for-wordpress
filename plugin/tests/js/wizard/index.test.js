/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { render } from '@testing-library/react';

/**
 * Internal dependencies
 */
import Wizard from '../../../assets/src/wizard/components';

describe( 'Onboarding Wizard', () => {
	beforeAll( () => {
		global.materialDesignWizard = {
			pagesUrl: 'http://example.com/wp-admin/edit.php?post_type=page',
			settingsUrl: 'http://example.com/wp-admin/plugins.php',
			assetsPath: 'http://example.com/',
			placeholderSmall: 'http://example.com/image-small.png',
			placeholderImage: 'http://example.com/image.png',
			themeStatus: 'ok',
			nonce: 'test-nonce',
			restPath:
				'http://example.com/index.php?rest_route=/material-design/v1/importer/',
		};
	} );

	it( 'matches snapshot', () => {
		const { container } = render( <Wizard /> );

		expect( container ).toMatchSnapshot();
	} );
} );
