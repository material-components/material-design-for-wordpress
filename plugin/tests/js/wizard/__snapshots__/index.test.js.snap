// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding Wizard matches snapshot 1`] = `
<div>
  <div
    class="material-wizard mdc-layout-grid"
  >
    <div
      class="mdc-layout-grid__inner"
    >
      <div
        class="mdc-layout-grid__cell mdc-layout-grid__cell--span-6 mdc-layout-grid__cell--align-middle"
      >
        <div
          class="mdc-layout-grid__inner"
        >
          <div
            class="mdc-layout-grid__cell mdc-layout-grid__cell--span-1 mdc-layout-grid__cell--align-middle"
          >
            <svg
              fill="none"
              height="41"
              width="41"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M34.345 34.673c3.59-3.619 5.81-8.596 5.81-14.081 0-5.486-2.22-10.463-5.81-14.081a.763.763 0 00-.108-.109c-3.62-3.59-8.596-5.81-14.082-5.81A19.928 19.928 0 006.074 6.404a.64.64 0 00-.106.107A19.928 19.928 0 00.155 20.59c0 5.488 2.221 10.465 5.813 14.082.031.04.066.075.106.106a19.928 19.928 0 0014.081 5.813c5.486 0 10.463-2.221 14.081-5.81a.895.895 0 00.109-.109zM7.215 9.929l11.8 23.602h-11.8V9.93zm.436-2.277H32.66L20.155 32.66 7.651 7.652zM33.095 9.93v23.602h-11.8l11.8-23.602zM34.5 32.406V8.777a18.505 18.505 0 014.246 11.815c0 4.483-1.594 8.6-4.246 11.814zM31.97 6.246H8.34a18.507 18.507 0 0111.815-4.248c4.484 0 8.6 1.596 11.815 4.248zM5.81 8.777v23.63a18.507 18.507 0 01-4.248-11.815c0-4.483 1.595-8.6 4.247-11.815zm2.53 26.16h23.63a18.505 18.505 0 01-11.815 4.247c-4.483 0-8.6-1.594-11.814-4.246z"
                fill="#000"
              />
            </svg>
          </div>
          <div
            class="mdc-layout-grid__cell mdc-layout-grid__cell--span-11 mdc-layout-grid__cell--align-middle"
          >
            <h2
              class="mdc-typography--headline5"
            >
              Material Design for WordPress
            </h2>
          </div>
        </div>
      </div>
      <div
        class="mdc-layout-grid__cell mdc-layout-grid__cell--span-6 mdc-layout-grid__cell--align-middle"
      >
        <ul
          class="material-wizard__progress"
        >
          <li
            class="active"
          >
            <span
              class="material-wizard__progress-dot"
            />
            <span
              class="material-wizard__progress-check"
            >
              <svg
                fill="none"
                height="21"
                width="21"
                xmlns="http://www.w3.org/2000/svg"
              >
                <mask
                  height="12"
                  id="a"
                  maskUnits="userSpaceOnUse"
                  width="16"
                  x="3"
                  y="5"
                >
                  <path
                    d="M8.313 13.82l-3.475-3.474-1.184 1.175 4.659 4.658 10-10-1.175-1.175-8.825 8.817z"
                    fill="#fff"
                  />
                </mask>
                <g
                  mask="url(#a)"
                >
                  <path
                    d="M.983.592h20v20h-20z"
                    fill="#fff"
                  />
                </g>
              </svg>
            </span>
          </li>
          <li
            class=""
          >
            <span
              class="material-wizard__progress-dot"
            />
            <span
              class="material-wizard__progress-check"
            >
              <svg
                fill="none"
                height="21"
                width="21"
                xmlns="http://www.w3.org/2000/svg"
              >
                <mask
                  height="12"
                  id="a"
                  maskUnits="userSpaceOnUse"
                  width="16"
                  x="3"
                  y="5"
                >
                  <path
                    d="M8.313 13.82l-3.475-3.474-1.184 1.175 4.659 4.658 10-10-1.175-1.175-8.825 8.817z"
                    fill="#fff"
                  />
                </mask>
                <g
                  mask="url(#a)"
                >
                  <path
                    d="M.983.592h20v20h-20z"
                    fill="#fff"
                  />
                </g>
              </svg>
            </span>
          </li>
          <li
            class=""
          >
            <span
              class="material-wizard__progress-dot"
            />
            <span
              class="material-wizard__progress-check"
            >
              <svg
                fill="none"
                height="21"
                width="21"
                xmlns="http://www.w3.org/2000/svg"
              >
                <mask
                  height="12"
                  id="a"
                  maskUnits="userSpaceOnUse"
                  width="16"
                  x="3"
                  y="5"
                >
                  <path
                    d="M8.313 13.82l-3.475-3.474-1.184 1.175 4.659 4.658 10-10-1.175-1.175-8.825 8.817z"
                    fill="#fff"
                  />
                </mask>
                <g
                  mask="url(#a)"
                >
                  <path
                    d="M.983.592h20v20h-20z"
                    fill="#fff"
                  />
                </g>
              </svg>
            </span>
          </li>
        </ul>
      </div>
      <div
        class="mdc-layout-grid__cell mdc-layout-grid__cell--span-12"
      >
        <div
          class="material-wizard__content mdc-layout-grid__inner"
        >
          <div
            class="mdc-layout-grid__cell--span-12"
          >
            <div
              class="material-wizard__card mdc-layout-grid__inner mdc-layout-grid__cell--align-middle"
            >
              <div
                class="mdc-layout-grid__cell mdc-layout-grid__cell--span-5"
              >
                <img
                  alt=""
                  src="http://example.com/welcome.png"
                />
              </div>
              <div
                class="mdc-layout-grid__cell mdc-layout-grid__cell--span-7 mdc-layout-grid__cell--align-middle"
              >
                <h3
                  class="mdc-typography--headline3"
                >
                  Start building
                </h3>
                <p>
                  Material Design for WordPress lets you use Material Components and custom styles in your WordPress site.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mdc-layout-grid__cell mdc-layout-grid__cell--span-12 material-wizard__navigation"
      >
        <div
          class="mdc-layout-grid__inner"
        >
          <div
            class="mdc-layout-grid__cell mdc-layout-grid__cell--span-7"
          >
            <a
              class="mdc-button material-wizard__close"
              href="http://example.com/wp-admin/plugins.php"
            >
              <span
                class="mdc-button__label"
              >
                Close
              </span>
            </a>
          </div>
          <div
            class="mdc-layout-grid__cell mdc-layout-grid__cell--span-5"
          >
            <div
              class="mdc-layout-grid__inner"
            >
              <div
                class="mdc-layout-grid__cell mdc-layout-grid__cell--span-6"
              />
              <div
                class="mdc-layout-grid__cell mdc-layout-grid__cell--span-6"
              >
                <button
                  class="mdc-button material-wizard__next mdc-button--raised"
                >
                  <span
                    class="mdc-button__label"
                  >
                    Next Step
                  </span>
                  <i
                    aria-hidden="true"
                    class="material-icons mdc-button__icon trailing-icon"
                  >
                    navigate_next
                  </i>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
